state/core_access.go

```diff
@@ -158,17 +158,29 @@ func (ca *CoreAccessor) constructSignedTx(
 
 func (ca *CoreAccessor) SubmitPayForBlob(
 	ctx context.Context,
-	nID namespace.ID,
-	data []byte,
+	nIDs []namespace.ID,
+	data [][]byte,
 	fee Int,
 	gasLim uint64,
 ) (*TxResponse, error) {
-	b := &apptypes.Blob{NamespaceId: nID, Data: data, ShareVersion: uint32(appconsts.DefaultShareVersion)}
+	if len(nIDs) != len(data) {
+		return nil, fmt.Errorf("state: amounts of nids=%d and data=%d are not equal", len(nIDs), len(data))
+	}
+
+	blobs := make([]*apptypes.Blob, len(data))
+	for i, share := range data {
+		blobs[i] = &apptypes.Blob{
+			NamespaceId:  nIDs[i],
+			Data:         share,
+			ShareVersion: uint32(appconsts.DefaultShareVersion),
+		}
```

Comment:
I think we should just pass the Blob type in the method. Mainly because we have to allow our users to specify share versions and extending one with one more param is a no go